### PR TITLE
[release/v2.8] Change `git.rancher.io` clone to `github.com` repos only during Dockerfile build

### PR DESCRIPTION
**Pull request by macedogm**
## Issue: 

This PR helps solve, at least partially, https://github.com/rancher/rancher/issues/43065.
 
## Problem

Change in `package/Dockerfile` and `tests/v2/codecoverage/package/Dockerfile` a total of 06 `git clone` operations done against `git.rancher.io` that might fail intermittently due to 502 errors in `git.rancher.io` backend. The operations are now done against the main repos in `github.com` during only the Dockerfile build process, and then it changes the `origin` URL back to `git.rancher.io` to avoid possible problems during Rancher's runtime execution. It's expected that this will help reduce a portion of the unnecessary load in `git.rancher.io` due to all the running PRs that will stop pulling directly from it.

Note: this doesn't change the URL inside Rancher's Go code. This can be done later and might require some extra logic.
 
### Automated Testing

* Test types added/modified:
    * Unit
    * Integration (Go Framework)
    * Integration (v2prov Framework)
    * Validation (Go Framework)
    * Other - Explain: _EXPLAIN_
    * None
    * _REMOVE NOT APPLICABLE BULLET POINTS ABOVE_
* If "None" - Reason: _EXPLAIN THE REASON_
  
* If "None" - GH Issue/PR: _LINK TO GH ISSUE/PR TO ADD TESTS_

Summary: _TODO_

### Regressions Considerations

_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

File: package/Dockerfile [hunk 1]
```diff
@@ -64,15 +64,28 @@ ENV CATTLE_CSP_ADAPTER_MIN_VERSION=103.0.0+up3.0.0
...
+    mkdir -p /var/lib/rancher-data/local-catalogs/v2
+
+RUN git clone -b $CATTLE_SYSTEM_CHART_DEFAULT_BRANCH --depth 1 https://github.com/rancher/system-charts /var/lib/rancher-data/local-catalogs/system-library && \
     git clone -b master --depth 1 https://github.com/rancher/charts /var/lib/rancher-data/local-catalogs/library && \
     git clone -b master --depth 1 https://github.com/rancher/helm3-charts /var/lib/rancher-data/local-catalogs/helm3-library
 
+# Charts need to be copied into the sha256 value of git url computed in https://github.com/rancher/rancher/blob/5ebda9ac23c06e9647b586ec38aa51cc9ff9b031/pkg/catalogv2/git/download.go#L102 to create a unique folder per url
+RUN git clone -b $CATTLE_CHART_DEFAULT_BRANCH --depth 1 https://github.com/rancher/charts /var/lib/rancher-data/local-catalogs/v2/rancher-charts/4b40cac650031b74776e87c1a726b0484d0877c3ec137da0872547ff9b73a721/ && \
+    cd /var/lib/rancher-data/local-catalogs/v2/rancher-charts/4b40cac650031b74776e87c1a726b0484d0877c3ec137da0872547ff9b73a721/ && \
+    git remote set-url origin https://git.rancher.io/charts && \
```
Comment (mattfarina) [inline package/Dockerfile:76]: FYI, you can set the remote URL without changing directory. Something like...

```
git --git-dir=/var/lib/rancher-data/local-catalogs/v2/rancher-charts/4b40cac650031b74776e87c1a726b0484d0877c3ec137da0872547ff9b73a721/.git remote set-url origin https://git.rancher.io/charts
```

All git commands allow you to point to the `.git` directory to run against.

Comment (macedogm) [reply package/Dockerfile:76]: Thanks, Matt! This command is really handy. I updated the PR.

**Comment by pjbgf** [comment]
As per [comment on 2.9](https://github.com/rancher/rancher/pull/43846#discussion_r1434102216), it may be easier to keep all commands as-is, and just add before the first clone: `git config --global url."https://github.com/rancher/".insteadOf https://git.rancher.io/` so there is no need to reset remotes. At the end it can be undone by unsetting it globally.

**Review by colstrom — CHANGES_REQUESTED** [review]
@pjbgf's solution gets my vote as discussed [here](https://github.com/rancher/rancher/pull/43846#discussion_r1450831366).

**Review by pjbgf — APPROVED** [review]

**Review by colstrom — APPROVED** [review]
LGTM

**Pull request merged**